### DISCUSSION TITLE
A modifier key in taiko editor to switch beat snap to triplets while held

### DISCUSSION THREAD

**Discussion by Firmatorenio** [ppy/osu#20870] [Ideas] [answered]
Seeing as switching between 1/4 and 1/6 in lazer editor is an absolute pain (back in stable it was just shifting a slider, in lazer you have to aim at a tiny button to switch common snap to triplets which resets you to 1/1 every time, from where you have to aim at another tiny button and click it twice.. repeat upon switching back), I propose holding some button (possibly <kbd>Alt</kbd> because there's no concept of distance snap in taiko) _specifically in taiko editor_ (seeing as it's exceedingly common there) make the timeline ticks snap in the triplet variant of the current used common snap and vice versa (so if 1/2 is selected, holding that button would temporarily switch that snap to 1/3, 1/4 would switch to 1/6, 1/8 would switch to 1/12 and vice versa).

Yes I know I could _technically_ just always be in 1/12 in order to be able to snap 1/4 and 1/6 interchangably (and use shift+scroll to offset the slow scrolling through the timeline), but it really makes one a lot more prone to error putting the notes in the wrong snaps and adds a lot of noise on the timeline. Personally, I would much rather have a simple solution seeing as stable had one.

Alternatively one could just make switching between triplets and common snaps remember what they were before and not reset to 1/1 every time.

**Comment by Doksha** [comment, marked as answer]
The editor itself just needs tons of change. All that needs to be brought back is the shift modifier so you can set the snap to whatever number you press with it (shift + 2 sets half notes, shift + 3 sets third notes, etc.)

**Reply by peppy** [reply]
https://github.com/ppy/osu/pull/20903
